src/eve/utils.py

```diff
@@ -33,57 +38,113 @@
     slugify,
     unwrap_text,
 )
+from boltons.typeutils import classproperty  # noqa: F401
+
+from ._typing import (
+    Any,
+    Callable,
+    Iterable,
+    Iterator,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    Union,
+)
+from .type_definitions import DELETE, NOTHING
+
+
+def filter_map(
+    func: Callable[..., Any], iterable: Iterable[Any], *, delete_sentinel: Any = DELETE
+) -> Iterator[Any]:
+    """Mapping function supporting elimination of items.
+
+    Args:
+        iterable: iterable object to be processed.
+        delete_sentinel: sentinel object which marks items to be deleted from the
+            the output (note that comparison is made by identity NOT by value).
+
+    """
+    for item in iterable:
+        result = func(item)
+        if result is not delete_sentinel:
+            yield result
 
-from . import typing
-from .typing import Any, Callable, Iterable, List, Mapping, Optional, Sequence, Set, Tuple, Union
 
+filter_map.DELETE = DELETE  # type: ignore
```

Comment:
I wonder what this does? It seems to have no effect at first glance.

Reply:
It does nothing, you're completely right. The idea was to plug the sentinel value to the function, so users can refer to it easily without using the global name. I think it makes sense but I don't have a strong preference so it can also be deleted. I added a docstring to make sure it is at least documented.

Reply:
I like the idea too.